### ISSUE TITLE
links generated from dataview tables point to incorrect urls

### ISSUE THREAD

**Issue by saezhe** [oleeskild/obsidian-digital-garden#64]
all links generated from dataview tables link to urls with the path `/notes/folder-1-folder-2-etc-file-name-md` as if the pages do not exist.
the links are also all less brightly highlighted as if the published pages aren't published.

i found this out when trying to make a sitemap page - that has a dataview query that pulls all current published pages to the site.
here is said sitemap page: [erosaezhe.netlify.app/9-extras/970-digital-garden-grounds/dg-sitemap/](https://erosaezhe.netlify.app/9-extras/970-digital-garden-grounds/dg-sitemap/)

example of one of the links:
![image](https://user-images.githubusercontent.com/45216469/182743313-7a7efc02-e371-43dc-b00e-cfeab4756bb8.png)
should link to this existing page:
[erosaezhe.netlify.app/2-zettels/bibasic-decimal-system/](https://erosaezhe.netlify.app/2-zettels/bibasic-decimal-system/)
instead links to:
[erosaezhe.netlify.app/notes/2-zettels-bibasic-decimal-system-md](https://erosaezhe.netlify.app/notes/2-zettels-bibasic-decimal-system-md)

**Comment by saezhe** [comment]
as a secondary follow up, there are probably custom tools i can employ in github to achieve a sort of navigation menu, but i'm trying to figure an in-house easy way that works for my obsidian as well at the moment.

and, i am also curious, what is the reason for dead links to direct to the alternate format urls, anyway? (with the `/notes/` prefix)
is it an old version of the plugin's url format that wasnt changed when working on a new version? or was it made that way on purpose?

**Comment by oleeskild** [comment]
Awesome, you found a bug 🎉 Thanks for reporting it!
It seems the issue was that some dataview queries adds an .md behind the filepath, which the digital garden template was not expecting. I've release a new template version, [1.14.1](https://github.com/oleeskild/digitalgarden/releases/tag/1.14.1), which should fix this issue 🤞.

You can update your version of the site template by going to Obsidian settings > digital garden > manage site template > create pr. You will then be presented with a link after about 20 seconds. Clicking this lets you complete the pull request and merge the latest changes into your repository.

Can you test if the new version fixes your problem or not, and report back on the results?

**Comment by oleeskild** [comment]
Regarding your second comment:
The reason the /notes/ is added is simply because the first version did not use folders to specify URL paths. All notes were simply put in a notes folder. Thus the default link is still /notes/{filename}, and it is only overwritten if the file it links to actually exists. 

The reason it was prefixed with notes was that initially I thought it would be a cool idea to leverage Git's version history to display previous versions of a note, and show how the note grew over time. These notes would have a /versionednotes/ prefix. This has actually been implemented but I've disabled it, as I wanted to focus on implementing other features first. 

To summarize: As you correctly guessed, it's a relic from an old version of the plugin that I haven't bothered to change.

**Comment by saezhe** [comment]
wow, thank you for the amazing accommodation! i'm about to try a PR.

results:
yes, the version preview works as i envisioned. thank you so much!

i asked about the dead notes urls because when i noticed them (for non-anomalous dead links) i thought it was a bit odd. i would have expected the urls to have parity with each other. if you want to fix dead links' parity as well, i would like to request it as a feature, just as a low priority one.

**Comment by oleeskild** [comment]
Please do. If you could add a new issue describing what you want and why you want it that would be great.
